ansible_collections/arista/avd/docs/contribution/development-tooling.md

```diff
@@ -0,0 +1,74 @@
+<!--
+  ~ Copyright (c) 2023-2024 Arista Networks, Inc.
+  ~ Use of this source code is governed by the Apache License 2.0
+  ~ that can be found in the LICENSE file.
+  -->
+
+# Development Tooling
+
+## Makefile
+<!-- TO DO - update!!! -->
+
+## Pre-commit hook
+
+[pre-commit](https://github.com/aristanetworks/avd/blob/devel/.pre-commit-config.yaml) can run standard hooks on every commit to automatically point out issues in code such as missing semicolons, trailing whitespace, and debug statements. Pointing these issues out before code review allows a code reviewer to focus on the architecture of a change while not wasting time with trivial style nitpicks.
+
+Repository implements the following hooks:
+
+- `trailing-whitespace`: Fix trailing whitespace. If found, the commit is stopped, and you must rerun the commit process.
+- `end-of-file-fixer`: Like `trailing-whitespace`, this hook fixes the wrong end of the file and stops your commit.
+- `check-yaml`: Checks that all YAML files are valid.
+- `check-added-large-files`: Check if no large file is included in the repository.
+- `check-merge-conflict`: Validate there is no `MERGE` syntax related to an invalid merge process.
+- `pylint`: Run Python linting with settings defined in [pylintrc](https://github.com/aristanetworks/avd/blob/devel/pylintrc).
+- `yamllint`: Validate all YAML files using configuration from [yamllintrc](https://github.com/aristanetworks/avd/blob/devel/.github/yamllintrc).
+- `ansible-lint`: Validate YAML files with Ansible proven practices, patters, and behaviors.
+- `Flake8`: Style guide enforcement for Python code base.
+- `markdownlint-cli`: Validates markdown files for common errors as referenced [here](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md).
+
```

Comment:
j2lint also

Reply:
we have decided to simply point to the pre-commit-config.yaml file so we don't have to maintain a list!